### PR TITLE
[Reader] Add more Google Analytics for Reader

### PR DESCRIPTION
**Pull request by NickHeiner**
Connect #2761.

### CODE REVIEW

File: app/views/reader/documents/show.html.erb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 <% content_for :full_page_content do %>
   <%= react_component("BaseContainer", props: {
+    singleDocumentMode: true,
```
Comment (NickHeiner) [inline app/views/reader/documents/show.html.erb]: Previously, we enter single document mode if `documents.length === 1`. This seems brittle to me, so I'm introducing an explicit new variable.

File: client/app/components/PdfSidebar.jsx [hunk 2]
```diff
@@ -43,10 +43,25 @@ export class PdfSidebar extends React.Component {
...
 
-  toggleKeyboardModal = () => {
-    this.setState((prevState) => ({
-      modal: !prevState.modal
-    }));
+  handleKeyboardModalClose = () => this.toggleKeyboardModal('modal-close-handler')
+  closeKeyboardModalFromButton = () => this.toggleKeyboardModal('modal-got-it-button')
+  openKeyboardModal = () => this.toggleKeyboardModal('view-shortcuts-button')
+
+  toggleKeyboardModal = (sourceLabel) => {
```
Comment (mmlumba) [inline client/app/components/PdfSidebar.jsx]: Where does the source label come from?

Comment (NickHeiner) [inline client/app/components/PdfSidebar.jsx]: We can see that it's an argument to this function. If we `grep` for usage of this function, we can see that it's used in the three lines above this point in the file:

image

From there, we can see where `sourceLabel` is defined. Does that help answer your question? I'm happy to follow up if there is more that I can explain.

**Comment by NickHeiner** [comment]
Thanks!

**Pull request merged**